sky/data/storage.py

```diff
@@ -766,6 +766,7 @@ def warn_for_git_dir(source: str):
 
     @classmethod
     def from_yaml_config(cls, config: Dict[str, Any]) -> 'Storage':
+        config = {k: v for k, v in config.items() if v is not None}
```

Comment:
Possible to move this into backend_utils.validate_schema since there are three places?

Reply:
Possible, but that would remove the possibility of us having nullable YAML fields in the future. I'll add a flag so we can decide whether or not we want to remove None values.

Reply:
I've added a flag so we can enable this for validation. The reason we don't want to do it for all cases is that it prevents us from actually having null values in the schema in the future.

In addition, a lot of the current code relies on the fact that if a key exists in the config dictionary, then the value is valid. E.g. if we allow None to be inside the config dictionary, then inside `storage.from_yaml_config`, `force_delete = config.pop('_force_delete', False)` would incorrectly set `force_delete` to None if the user specifies `force_delete` in the YAML without a value. Therefore, I also had to refactor the code such that it properly handle `None` config values.